### PR TITLE
Adds register feature using API

### PR DESCRIPTION
**Pull request by geekyd**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Enables registration using the new API. 

#### Changes proposed in this pull request:

- Adds user serializer for filtering the request.
- Post calls are now handled in the controller.

### CODE REVIEW

**Comment by niranjan94** [comment]
Also, override `willDestroy` method of the controller and unload the model

**Comment by geekyd** [comment]
@niranjan94 I have made some changes please review. Thanks

File: app/routes/register.js [hunk 1]
```diff
@@ -6,5 +6,10 @@ const { Route } = Ember;
 export default Route.extend(UnauthenticatedRouteMixin, {
   titleToken() {
     return this.l10n.t('Register');
+  },
+  model() {
+    return {
+      user: this.store.createRecord('user')
```
Comment (niranjan94) [inline app/routes/register.js]: Directly return the user record. No need of an object here.

File: app/controllers/register.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  actions: {
+    createUser() {
+      var user = this.get('model.user');
+      user.save()
+        .then(() => {
+          this.transitionToRoute('login');
+        })
+        .catch(data => {
+          const error = data.errors[0].status;
```
Comment (niranjan94) [inline app/controllers/register.js]: call this as `statusCode`

File: app/controllers/register.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    var user = this.get('model');
+    this.store.unloadRecord(user);
+  },
+
+  actions: {
+    createUser() {
+      var user = this.get('model');
+      user.save()
+        .then(() => {
+          this.transitionToRoute('login');
```
Comment (niranjan94) [inline app/controllers/register.js]: Also show a notification (using `notify`) that says 
`Welcome <user-name>. Please login to continue.`

**Pull request merged**